nvtabular/workflow/node.py

```diff
@@ -65,16 +67,105 @@ def selector(self, sel):
 
         self._selector = sel
 
+    @property
+    def dependency_schema(self):
+        schema = Schema()
+        for dep in self.dependencies:
+            if isinstance(dep, ColumnSelector):
+                schema += Schema(dep.names)
+            elif isinstance(dep, WorkflowNode):
+                schema += dep.output_schema
+            elif isinstance(dep, list):
+                for nested_dep in dep:
+                    if isinstance(nested_dep, ColumnSelector):
+                        schema += Schema(nested_dep.names)
+                    elif isinstance(nested_dep, WorkflowNode):
+                        schema += nested_dep.output_schema
```

Comment:
This is the same code twice; we ought to be able to write it as a recursive method